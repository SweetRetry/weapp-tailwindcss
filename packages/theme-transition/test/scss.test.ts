import { pathToFileURL } from 'node:url'
import path from 'pathe'
import { compile, compileString } from 'sass-embedded'
// import sassTrue from 'sass-true'
const importers = [
  {
    findFileUrl(url: string) {
      const p = path.resolve(__dirname, url)
      return new URL(
        pathToFileURL(p),
      )
    },
  },
]
describe('scss', () => {
  it('index', () => {
    const { css } = compile(path.resolve(__dirname, '../scss/index.scss'))
    expect(css).toMatchSnapshot()
  })

  it('compileString', () => {
    const { css } = compileString(`@use '../scss/mixins.scss' as M;
      @include M.theme-transition;`, {
      importers,
    })
    expect(css).toMatchSnapshot()
  })

  it('compileString case 0', () => {
    const { css } = compileString(`@use '../scss/mixins.scss' as M;
      @include M.theme-transition("[data-theme='dark']");`, {
      importers,
    })
    expect(css).toMatchSnapshot()
  })

  it('compileString case 1', () => {
    const { css } = compileString(`@use '../scss/mixins.scss' as M;
      @include M.theme-transition("[data-theme='dark']",2,3);`, {
      importers,
    })
    expect(css).toMatchSnapshot()
  })

  // it('sassTrue', () => {
  //   const sassFile = path.join(__dirname, 'test.scss')
  //   sassTrue.runSass({ describe, it }, sassFile, {
  //     silenceDeprecations: ['legacy-js-api'], // , 'color-4-api'],
  //     importers: {
  //       findFileUrl(url: string) {
  //         const p = path.resolve(__dirname, url)
  //         return new URL(
  //           pathToFileURL(p),
  //         )
  //       },
  //     },
  //   })
  // })
})
