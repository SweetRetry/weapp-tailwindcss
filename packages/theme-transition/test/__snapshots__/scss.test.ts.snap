// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`scss > compileString 1`] = `
"::view-transition-old(root),
::view-transition-new(root) {
  mix-blend-mode: normal;
  animation: none;
}

::view-transition-old(root) {
  z-index: 1;
}

::view-transition-new(root) {
  z-index: 2147483646;
}

.dark::view-transition-old(root) {
  z-index: 2147483646;
}

.dark::view-transition-new(root) {
  z-index: 1;
}"
`;

exports[`scss > compileString case 0 1`] = `
"::view-transition-old(root),
::view-transition-new(root) {
  mix-blend-mode: normal;
  animation: none;
}

::view-transition-old(root) {
  z-index: 1;
}

::view-transition-new(root) {
  z-index: 2147483646;
}

[data-theme=dark]::view-transition-old(root) {
  z-index: 2147483646;
}

[data-theme=dark]::view-transition-new(root) {
  z-index: 1;
}"
`;

exports[`scss > index 1`] = `
"::view-transition-old(root),
::view-transition-new(root) {
  mix-blend-mode: normal;
  animation: none;
}

::view-transition-old(root) {
  z-index: 1;
}

::view-transition-new(root) {
  z-index: 2147483646;
}

.dark::view-transition-old(root) {
  z-index: 2147483646;
}

.dark::view-transition-new(root) {
  z-index: 1;
}"
`;
